### PR TITLE
fix: the crash when the pageSize of pagination in List is undefined

### LINKED ISSUE #9496: Calendar组件选择非当前月的日期 onPanelChange 不能触发的问题
当我点击日历中上个月或者下个月中的日期时，日期面板已经发生变化，但是没有触发onPanelChange 方法，会导致页面数据没有加载，带来了极差的用户体验，虽然我可以用onSelect方法来解决这个问题，但是我认为官方应该重视这个问题！！！！！！

### PR DESCRIPTION
**Pull request by BoyYangzai**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
- fix #9496

### 📝 Changelog
Fix the crash when the pageSize of pagination in List is undefined  
修复List中分页的 pageSize 为 undefined 时崩溃 

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Fix the crash when the pageSize of pagination in List is undefined       |
| 🇨🇳 Chinese |      修复List中分页的 pageSize 为 undefined 时崩溃     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/list/index.tsx [hunk 1]
```diff
@@ -194,8 +194,8 @@ function List<T>({
     ...defaultPaginationProps,
     total: dataSource.length,
     current: paginationCurrent,
-    pageSize: paginationSize,
     ...(pagination || {}),
+    pageSize: paginationSize,
```
Comment (yoyo837) [inline components/list/index.tsx:198]: 这解法感觉不太对吧?

Comment (BoyYangzai) [reply components/list/index.tsx:198]: 看了下原因，是因为传下去的 `pagination` 是个 `{pagesize：undefined} `,后面的 `...(pagination || {})` 中产生的 `pageSize` 把前面正常逻辑的覆盖了

Comment (yoyo837) [reply components/list/index.tsx:198]: 这个应该由业务层自己规避吧?

Comment (BoyYangzai) [reply components/list/index.tsx:198]: > 这个应该由业务层自己规避吧?

我敲，我想了下,这样改确实不妥，我等会再改一下，这里只需要处理一下 `...(pagination || {})` ，不需要更改顺序

**Comment by BoyYangzai** [comment]
@yoyo837 我把逻辑做了下优化，以及缺少一个 `pageize` 的类型提示

File: components/list/index.tsx [hunk 2]
```diff
@@ -50,7 +50,7 @@ export interface ListProps<T> {
   itemLayout?: ListItemLayout;
   loading?: boolean | SpinProps;
   loadMore?: React.ReactNode;
-  pagination?: PaginationConfig | false;
```
Comment (yoyo837) [inline components/list/index.tsx:53]: 这里为啥删除, `pagination` 支持 `false` 是预期的

File: components/pagination/Pagination.tsx [hunk 3]
```diff
@@ -25,6 +25,7 @@ export type PaginationPosition = 'top' | 'bottom' | 'both';
 
 export interface PaginationConfig extends PaginationProps {
   position?: PaginationPosition;
+  pageSize?: number;
```
Comment (yoyo837) [inline components/pagination/Pagination.tsx:28]: `pageSize` 在继承的上面是存在的

File: components/list/index.tsx (hunk 2 above)
Comment (BoyYangzai) [reply components/list/index.tsx:53]: > 这里为啥删除, `pagination` 支持 `false` 是预期的

类型提示有问题，这里为了一个边界case修改太多不太值得了，我先关闭这个pr了

**Pull request closed without merging**